src/Data/Fin/Subset/Properties.agda

```diff
@@ -576,15 +575,11 @@ Lift? P? p = decFinSubset (_∈? p) (λ {x} _ → P? x)
 -- Other
 
 anySubset? : ∀ {n} {P : Subset n → Set} → Decidable P → Dec (∃ P)
-anySubset? {zero}  P? with P? []
-... | yes P[] = yes (_ , P[])
-... | no ¬P[] = no (λ {([] , P[]) → ¬P[] P[]})
-anySubset? {suc n} P? with anySubset? (P? ∘ (inside ∷_))
-... | yes (_ , Pp) = yes (_ , Pp)
-... | no  ¬Pp      with anySubset? (P? ∘ (outside ∷_))
-...   | yes (_ , Pp) = yes (_ , Pp)
-...   | no ¬Pp'      = no λ
-  { (inside  ∷ p , Pp)  → ¬Pp  (_ , Pp)
-  ; (outside ∷ p , Pp') → ¬Pp' (_ , Pp')
-  }
+anySubset? {zero}  P? = map′ ([] ,_) (λ { ([] , P[]) → P[] }) (P? [])
+anySubset? {suc n} P? =
+  map′ [ Product.map _ id , Product.map _ id ]′
+       (λ { (outside ∷ p , Pp) → inj₂ (p , Pp)
+          ; ( inside ∷ p , Pp) → inj₁ (p , Pp)
+          })
```

Comment:
Could we split the second direction out into it's own lemma or a `where` clause?